src/shared/helpers/navigation.tsx

```diff
@@ -0,0 +1,137 @@
+import * as H from 'history';
+import { TFunction } from 'i18next';
+import { isNil, kebabCase, sortBy } from 'lodash-es';
+import queryString from 'query-string';
+import React, { FunctionComponent } from 'react';
+import { match } from 'react-router';
+import { NavLink } from 'react-router-dom';
+
+import { Dropdown, DropdownButton, DropdownContent, Icon } from '@viaa/avo2-components';
+
+import LoginOptionsDropdown from '../../authentication/components/LoginOptionsDropdown';
+import PupilOrTeacherDropdown from '../../authentication/components/PupilOrTeacherDropdown';
+import { APP_PATH } from '../../constants';
+import { AppContentNavElement } from '../services/navigation-items-service';
+import { NavigationItem } from '../types';
+import { buildLink } from './link';
+
+const NAVIGATION_COMPONENTS: { [componentLabel: string]: FunctionComponent<any> } = {
+	'<PupilOrTeacherDropdown>': PupilOrTeacherDropdown,
+	'<LoginOptionsDropdown>': LoginOptionsDropdown,
+};
+
+export function getLocation(navItem: AppContentNavElement, t: TFunction): string {
+	if (!isNil(navItem.content_id)) {
+		// Link to content block page
+		return `/${navItem.content_id}/${kebabCase(navItem.label)}`;
+	} else if (navItem.external_link) {
+		return navItem.external_link;
+	} else {
+		console.error('Failed to generate navigation link for navigation item', { navItem });
+		return buildLink(
+			APP_PATH.ERROR,
+			{},
+			queryString.stringify({
+				message: t('De pagina voor dit navigatie item kon niet worden gevonden'),
+				icon: 'search',
+			})
+		);
+	}
+}
+
+export function mapNavElementsToNavigationItems(
+	navItems: AppContentNavElement[],
+	history: H.History,
+	location: H.Location,
+	match: match<any>,
+	t: TFunction
+): NavigationItem[] {
+	return sortBy(navItems, 'position').map(
+		(navItem: AppContentNavElement): NavigationItem => {
+			const location: string = getLocation(navItem, t);
+			if (NAVIGATION_COMPONENTS[location]) {
```

Comment:
Jij kiest of je dit leesbaarder vindt of niet.
```
		const navigationDefault = {
			label: navItem.label,
			icon: navItem.icon_name,
			key: `nav-item-${navItem.id}`,
		}

		const Component = NAVIGATION_COMPONENTS[location];

		return Component ? { // Show component when clicking this nav item
				...navigationDefault,
				component: <Component history={history} location={location} match={match} />,
			} : { // Navigate to link
				...navigationDefault,
				location: getLocation(navItem, t),
				target: navItem.link_target,
			}
```

Reply:
niet echt nee